src/components/TodoModal/TodoModal.tsx

```diff
@@ -1,42 +1,73 @@
-import React from 'react';
+import React, { useEffect, useState } from 'react';
 import { Loader } from '../Loader';
+import { getUser } from '../../api';
+import { User } from '../../types/User';
+import { useAppDispatch, useAppSelector } from '../../app/hooks';
+import { actions as currentTodoActions } from '../../features/currentTodo';
 
 export const TodoModal: React.FC = () => {
+  const [user, setUser] = useState<User | null>(null);
+  const [loading, setLoading] = useState(true);
+
+  const dispatch = useAppDispatch();
+  const currentTodo = useAppSelector(state => state.currentTodo);
+
+  const removeCurrentTodo = () => dispatch(currentTodoActions.removeTodo());
+
+  if (currentTodo) {
+    getUser(currentTodo.userId).then(setUser);
+  }
+
+  useEffect(() => {
+    setTimeout(() => {
+      if (currentTodo) {
+        setLoading(false);
+      }
+    }, 1000);
+  }, [currentTodo]);
```

Comment:
you need to fetch the user in useEffect and after that call `setLoading(false);`